### PR TITLE
Stop initialising all navhost fragments at start 

### PR DESCRIPTION
**Pull request by VictorAlbertos**
This PR initializes lazily the NavController of each fragment tab. As a bonus, it removes the limitation of not being able to use the `initialSelectedTabId` when launched from a deep link.

Associated issue: https://github.com/cookpad/BottomNavWatson/issues/5

### CODE REVIEW

**Review by felipepedroso — APPROVED** [review]
LGTM :rocket:

File: bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt [hunk 1]
```diff
@@ -36,96 +32,90 @@ fun BottomNavigationView.setupWithNavController(
...
-                                    detach(fragmentManager.findFragmentByTag(firstFragmentTag)!!)
-                                }
+                // Commit a transaction that cleans the back stack and adds the initial fragment
+                // to it, creating the fixed started destination.
+                fragmentManager.beginTransaction()
+                    .attach(selectedFragment)
+                    .setPrimaryNavigationFragment(selectedFragment)
+                    .apply {
+                        // Detach all other Fragments
+                        tabIdToTagMap.forEach { _, fragmentTagIter ->
```
Comment (mbarta) [inline bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt]: [nit][naming] what does the `Iter` suffix mean? I'd leave it out :grimacing:

**Review by mbarta — APPROVED** [review]
Nice :rocket: It even seems to have simplified the code nicely :)
I just added an annoying nit about naming, sorry about that :)

File: bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt (hunk 1 above)
Comment (VictorAlbertos) [reply bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt]: Copy/pasted from the original Google sources, I'll rename it 👍 But I guess it scopes the naming to the iterable item of the forEach loop

Comment (VictorAlbertos) [reply bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt]: > Nice 🚀 It even seems to have simplified the code nicely :)

Yes, now I can completely understand the code xD

**Comment by VictorAlbertos** [comment]
Thanks for the review!

**Pull request merged**